### ISSUE TITLE
Autosubmit wrapper twin experiments

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#589]
In GitLab by @mcastril on Sep 16, 2020, 12:48

Hi @dbeltrankyl . We will coordinate the runs for the wrapper experiment here. 

I will run `a34n`, which is the wrapped version. You would run a copy of `a34o`, which is the non-wrapped.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 16, 2020, 12:49

Could you start with copying `a34o` and changing the credentials?

We will use Autosubmit 3.12.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 16, 2020, 18:27

@mcastril , 

Sorry, I just noticed that my mail notification didn't work today and I didn't notice that there was this issue on git due being busy with another issue. Can we start it tomorrow?

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 17, 2020, 11:01

Hi @mcastril ,

I'm setting up the experiment , which queue debug or bsc_es?

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 17, 2020, 13:19

Hi Daniel, this is high resolution, so the SIMs won't fit in debug. The rest of the jobs fit so we will use debug.

You can look at my jobs.conf (a34n).

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 17, 2020, 15:06

I'm running it, the expid is a358

edit: ok it is not running, one moment

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 17, 2020, 15:15

Hi @mcastril,

I removed the nctime job, but it is causing crashes on local_setup


```
Job SIM
    ERROR: In a production experiment (PRODUCTION_EXP = TRUE in proj) you must define: NCTIME
    ERROR: Jobs dependency: SIM-1 is not in SIM job dependencies
    ERROR: Jobs dependency: NCTIME-3 is not in SIM job dependencies

```

 Should I enable it?

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 17, 2020, 16:20

Please, set PRODUCTION_EXP = FALSE. Apologies, I didn't try it after removing NCTIME.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 17, 2020, 16:40

Thanks, it is running now

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2020, 08:51

Hi @mcastril ,

Seems that the first sim crashed

/esarchive/autosubmit/a358/tmp/LOG_a358/a358_19931101_fc0_1_SIM.20200917171914.out

/esarchive/autosubmit/a358/tmp/LOG_a358/a358_19931101_fc0_1_SIM.20200917171914.err

```
MPL_BUFFER_METHOD:  2   128000000
 COSP SIMULATOR NOT REQUESTED
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 56165 RUNNING AT s22r1b36
=   EXIT CODE: 7
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
   Intel(R) MPI Library troubleshooting guide:
      https://software.intel.com/node/561764
===================================================================================

```

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 18, 2020, 10:00

It seems it was a node failure. Do you know if it hit the walllclock time by chance?

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 18, 2020, 10:01

Strange... it didn't happen when I ran it last week. So it seems something totally random.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2020, 10:12

Hi  @mcastril ,

I don't think it reaches max wallclock which is set to 7:00
```
2020-09-17 17:19:20 
           17:24:55 -> running
           20:59:11 -> failed time
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 18, 2020, 10:39

Hi @dbeltrankyl . Could you check the node memory usage in the Userportal? And please ask to Operations in parallel, giving the JOBID... I am afraid this was a faulty node but I'd like to be sure.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2020, 11:16

Hi ,

I'm unable to find the SIM job in hpcportal, only INI and REMOTE

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 18, 2020, 12:02

Deleted runtime and running again.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 18, 2020, 16:31

Thanks @dbeltrankyl . Keep it going as long as it does not fail and next week we'll try to start the real experiment.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 22, 2020, 18:20

@dbeltrankyl it seems that you have had an issue with SIM_1. But you were already running SIM_3 ...

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 23, 2020, 10:51

running again

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 29, 2020, 17:52

Hi @dbeltrankyl, I realized that we didn't remove the unnecessary IMPI flags as it was intended in this experiment. This should make the SIMs have a more constant and faster performance (they should fit in 7 hours anyway, but this will improve the efficiency).

Could you copy my:

`/esarchive/autosubmit/a34n/proj/auto-ecearth3/sources/runtime/classic/platform/bsc-marenostrum4.cfg.tmpl`

To your file in the same relative path? And start again from LOCAL_SETUP. Sorry and thanks.

PS: You wouldn't need to remove the whole remote directory, with the runtime should be ok.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 30, 2020, 13:05

Hi @mcastril ,

Running again with the new template.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 16, 2020, 17:46

Hi @dbeltrankyl . My experiment has gone fine, not counting queuing times. It yours is also ok we can start next week.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 20, 2020, 11:42

@dbeltrankyl tomorrow we will coordinate ourselves to start the experiments at 10am approx.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 26, 2020, 19:07

Both experiments seem to go fine. 

a34n: Ending first wrapper, 6th SIM.
a358: 4th SIM QUEUING.

With these queuing times, the situation is more favourable to wrappers.

**Comment by kinow** [comment]
In GitLab by @mcastril on Oct 28, 2020, 14:45

Bad news, my experiment got a failed SIM because of a random MPI error... and yours was stopped today because of the nord3 outage. If it had been within a few hours we could just resume, but it's not the case.

So I think the best option is to start again... but you could try first the migrate, to make the most of this stop at least.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Oct 28, 2020, 15:54

Okay , I'll do the migrate first then.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 22, 2021, 12:53

Experiments resumed:

https://earth.bsc.es/autosubmitapp/experiment/a358  
https://earth.bsc.es/autosubmitapp/experiment/a34n

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 18, 2021, 17:02

```
146 of 147 jobs remaining (17:02)
Job a34n_SYNCHRONIZE is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 18, 2021, 17:04

```
145 of 147 jobs remaining (17:04)
Job a34n_REMOTE_SETUP is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 09:03

```
143 of 147 jobs remaining (09:02)
Job a358_19931101_fc0_1_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 19, 2021, 15:55

As commented in slack: 

Seems that a segmentation fault happened, autosubmit did the retry automatically and is still queuing: 

```
143 of 147 jobs remaining (15:53)
Job a358_19931101_fc0_1_SIM is QUEUING (Priority)
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 19, 2021, 16:49

```
141 of 147 jobs remaining (16:48)
Job a34n_19931101_fc0_1_SAVEIC is QUEUING (Priority)
Wrapper job a34n_ASThread_16136700995807_2400_6 is RUNNING
Job a34n_19931101_fc0_2_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 22, 2021, 09:14

```
129 of 147 jobs remaining (09:13)
Job a358_19931101_fc0_4_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:14

Thanks @dbeltrankyl for checking. My experiment has gone faster:

```
101 of 147 jobs remaining (10:12)
Wrapper job a34n_ASThread_16139821221472_2400_4 is QUEUING
```

10 SIMs completed by now.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:19

> 10 SIMs completed by now.

No! I was wrong. The 2nd wrapper was submitted but failed two times in the 8th chunk.

This is not good because I'm afraid that the 2nd time failed because the first time wrote part of the restarts and we didn't set the NEMO flag to be able to overwrite. Going to look what produced the first failure, it seems a wallclock issue.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:20

```
2021-02-21 19:19:46,770 Job a34n_19931101_fc0_8_SIM is RUNNING
2021-02-21 19:19:46,770 Job a34n_19931101_fc0_9_SIM is SUBMITTED and waiting for dependencies
2021-02-21 19:19:51,094 Command find /gpfs/scratch/bsc32/bsc32252/a34n/LOG_a34n  -name \*a34n_19931101_fc0_8_SIM_COMPLETED in mn1.bsc.es successful with out message:
[ERROR] 2021-02-21 19:19:51,094 Job a34n_19931101_fc0_8_SIM inside wrapper a34n_ASThread_16138449257156_2400_3 is running for longer than its wallclock! Cancelling...
2021-02-21 19:19:51,094 Cancelling job with id 14371735
2021-02-21 19:19:56,470 Command scancel 14371735 in mn1.bsc.es successful with out message:
2021-02-21 19:20:03,203 Command find /gpfs/scratch/bsc32/bsc32252/a34n/LOG_a34n  -name \*a34n_19931101_fc0_8_SIM_COMPLETED in mn1.bsc.es successful with out message:
2021-02-21 19:20:03,203 No completed filed found, setting a34n_19931101_fc0_8_SIM to FAILED...
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:26

The job was a little slow, but not a lot (7 hours and it was writing restarts while the previous were taking 6:30 - 6:40). Didn't remember that in 3.12.0 there's not 20% margin for wallclock time.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:30

With these 2 errors (restarts and wallclock) I am afraid that is better to start over. My run will not go on if I don't remove the remote restarts but these experiments should be run without user intervention to produce a fair comparison.

@dbeltrankyl, could you set 

```
[bsc32252@dtransfer1 ctrl]$ grep -r clobber *
namelist.nemo.ref.sh:   ln_clobber    = .true.        !  clobber (overwrite) an existing file
```

in 

`/esarchive/autosubmit/${EXPID}/proj/auto-ecearth3/sources/runtime/classic/ctrl/namelist.nemo.ref.sh`

And I will increase the wallclock time to 7:30 to avoid issues with simulations having small delays.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 10:30

Even better, let's change the wallclock to 8:00 in both experiments.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 22, 2021, 10:57

Hi @mcastril ,

> @dbeltrankyl, could you set

done 
> Even better, let's change the wallclock to 8:00 in both experiments.

I already had the wallclock of SIM to 8:00h

Should I delete the run and prepare for start again?

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 13:46

Oh, I had it set to 7:00...

> Should I delete the run and prepare for start again?

Yes please, sorry for that. I hope it's the last one, or very close to that.

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 15:27

```
146 of 147 jobs remaining (15:27)
Job a34n_SYNCHRONIZE is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 22, 2021, 15:27

```
146 of 147 jobs remaining (15:27)
Job a358_SYNCHRONIZE is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 22, 2021, 17:32

```
143 of 147 jobs remaining (17:15)
Wrapper job a34n_ASThread_16140100118187_2400_6 is RUNNING
Job a34n_19931101_fc0_1_SIM started at 2021-02-22 17:15:17
Job a34n_19931101_fc0_1_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 23, 2021, 08:44

```
138 of 147 jobs remaining (08:43)
Job a358_19931101_fc0_2_SAVEIC is QUEUING (Priority)
Job a358_19931101_fc0_3_SIM is QUEUING (Priority)
Job a358_19931101_fc0_2_CMOROCE is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 23, 2021, 13:54

```
131 of 147 jobs remaining (13:53)
Job a34n_19931101_fc0_1_EARTHDIAGS is RUNNING
Wrapper job a34n_ASThread_16140100118187_2400_6 is RUNNING
Job a34n_19931101_fc0_4_SIM is RUNNING
Job a34n_19931101_fc0_3_SAVEIC is QUEUING (Priority)
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 24, 2021, 09:20

```
137 of 147 jobs remaining (09:18)
Job a358_19931101_fc0_2_SAVEIC is COMPLETED
Job a358_19931101_fc0_3_SIM is QUEUING (Priority)
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 24, 2021, 10:54

```
108 of 147 jobs remaining (10:54)
Wrapper job a34n_ASThread_16141527729634_2400_2 is QUEUING
```

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 24, 2021, 18:19

```
108 of 147 jobs remaining (18:17)
Wrapper job a34n_ASThread_16141527729634_2400_2 is RUNNING
Job a34n_19931101_fc0_7_SIM started at 2021-02-24 18:17:57
Job a34n_19931101_fc0_7_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Feb 26, 2021, 09:13

```
120 of 147 jobs remaining (09:08)
Job a358_19931101_fc0_5_CMOROCE is RUNNING
Job a358_19931101_fc0_3_POST is QUEUING (QOSMaxJobsPerUserLimit)
Job a358_19931101_fc0_5_SAVEIC is QUEUING (Priority)
Job a358_19931101_fc0_6_SIM is QUEUING (Priority)
```
The SIM 5 failed once, 
`can not find matching ssend. context=c8, rank=92, tag=0`

But it finished on the next try and now is on SIM 6

**Comment by kinow** [comment]
In GitLab by @mcastril on Feb 26, 2021, 11:19

Thanks @dbeltrankyl.

These errors are painful, I am doing some research on them. Finally, my SIM_8 went through. Oddly enough, I had two consecutive failures on SIM_8, which was the chunk that failed twice in the previous run...

```
95 of 147 jobs remaining (11:17)
Job a34n_19931101_fc0_8_SAVEIC is RUNNING
Wrapper job a34n_ASThread_16142685274268_2400_4 is RUNNING
Job a34n_19931101_fc0_9_SIM is RUNNING
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Mar 1, 2021, 09:34

It had some issues(auto recovered):

```
[WARNING] Command sacct -n -j 14548845 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_10_SIM completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_10_SIM in UNKNOWN status
[WARNING] Could not get start time for a358_19931101_fc0_10_SIM. Using current time as an approximation

```

Not only for SIM 10 recently but also for others jobs during this night

```
[WARNING] Command sacct -n -j 14548848 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_7_POST completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_7_POST in UNKNOWN status
[WARNING] Could not get start time for a358_19931101_fc0_7_POST. Using current time as an approximation
[WARNING] Command sacct -n -j 14548844 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_9_SAVEIC completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_9_SAVEIC in UNKNOWN status
[WARNING] Command sacct -n -j 14548845 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_10_SIM completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_10_SIM in UNKNOWN status
[WARNING] Could not get start time for a358_19931101_fc0_10_SIM. Using current time as an approximation
[WARNING] Command sacct -n -j 14548847 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_9_CMOROCE completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_9_CMOROCE in UNKNOWN status


95 of 147 jobs remaining (07:43)
[WARNING] Command sacct -n -j 14548848 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_7_POST completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_7_POST in UNKNOWN status
[WARNING] Could not get start time for a358_19931101_fc0_7_POST. Using current time as an approximation
[WARNING] Command sacct -n -j 14548844 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_9_SAVEIC completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_9_SAVEIC in UNKNOWN status
[WARNING] Command sacct -n -j 14548845 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_10_SIM completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_10_SIM in UNKNOWN status
[WARNING] Could not get start time for a358_19931101_fc0_10_SIM. Using current time as an approximation
[WARNING] Command sacct -n -j 14548847 -o "State" in mn1.bsc.es warning: sacct: error: slurm_persist_conn_open_without_init: failed to open persistent connection to slurmdbd.bsc.mn:6819: Connection refused
sacct: error: slurmdbd: Sending PersistInit msg: Connection refused
sacct: error: Problem talking to the database: Connection refused

[WARNING] Job a358_19931101_fc0_9_CMOROCE completion check failed. There is no COMPLETED file
[WARNING] Job a358_19931101_fc0_9_CMOROCE in UNKNOWN status
```




Current status:
```
91 of 147 jobs remaining (09:27)
Job a358_19931101_fc0_7_EARTHDIAGS is RUNNING
Job a358_19931101_fc0_10_SIM is QUEUING (Priority)
```

Now seems that it is okay
